### PR TITLE
Update ingestion metadata android v2 example

### PR DESCRIPTION
**Pull request by yuhao900914**

### CODE REVIEW

File: android/java/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/SetAmpliExtrasPlugin.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    @Override
+    public BaseEvent execute(@NonNull BaseEvent baseEvent) {
+        Map<String, Object> ampliExtra = new HashMap<>();
+        Map<String, String> ingestionMetadataValue = new HashMap<>();
+        ingestionMetadataValue.put("sourceName", "android-java-ampli");
+        ingestionMetadataValue.put("sourceVersion", "2.0.0");
+        Map<String, Map<String, String>> ingestionMetadataMap = new HashMap<>();
+        ingestionMetadataMap.put("ingestionMetadata", ingestionMetadataValue);
+        ampliExtra.put("ampli", ingestionMetadataMap);
+        baseEvent.setExtra(ampliExtra);
```
Comment (qingzhuozhen) [inline android/java/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/SetAmpliExtrasPlugin.java:41]: this is overwriting the value?

File: android/java/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/SetAmpliExtrasPlugin.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    public void setAmplitude(@NonNull com.amplitude.core.Amplitude amplitude) {
+        this.amplitude = amplitude;
+    }
+
+    @NonNull
+    @Override
+    public Type getType() {
+        return Type.Before;
```
Comment (qingzhuozhen) [inline android/java/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/SetAmpliExtrasPlugin.java:28]: The order is already tested?

Comment (yuhao900914) [reply android/java/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/SetAmpliExtrasPlugin.java:28]: Yes

File: android/java/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/SetAmpliExtrasPlugin.java (hunk 1 above)
Comment (yuhao900914) [reply android/java/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/SetAmpliExtrasPlugin.java]: Update this to merge original extra and ingestionMetadata

**Review by liuyang1520 — APPROVED** [review]

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**